src/common/ceph_context.cc

```diff
@@ -1042,7 +1042,7 @@ void CephContext::notify_pre_fork()
 
 void CephContext::notify_post_fork()
 {
-  ceph::spin_unlock(&_fork_watchers_lock);
+  std::lock_guard lg(_fork_watchers_lock);
```

Comment:
1. `_fork_watchers_lock` is `ceph::spinlock`-typed, **`private`** member of `CephContext`.
2. Apart `notify_post_fork()` it has only 2 other users:
```cpp
void CephContext::notify_pre_fork()
{
  {
    std::lock_guard lg(_fork_watchers_lock);
    for (auto &&t : _fork_watchers) {
      t->handle_pre_fork();
    }
  }
```
```cpp
  void register_fork_watcher(ForkWatcher *w) {
    std::lock_guard lg(_fork_watchers_lock);
    _fork_watchers.push_back(w);
  }
```

Per the reasoning above the fix LGTM!

Reply:
Thanks @rzarzynski. I think we need a tracker for this for backports - will resend with updated tracker link.